caramel/module/scripts/caramel.handlebars.js

```diff
@@ -513,14 +514,54 @@ engine('handlebars', (function () {
      */
     theme = function (page, contexts, js, css, code) {
         var file, template, path, area, blocks, helper, length, i, o, areas, block,
-            areaContexts, data, areaData, find, blockData, resolve, partials, findPartials, analyzePartials,
+            areaContexts, data, areaData, find, blockData, resolve, partials, findPartials, analyzePartials, testFunction,
             theme = caramel.theme(),
             meta = caramel.meta(),
-            xcd = meta.request.getHeader(caramelData);
+            xcd = meta.request.getHeader(caramelData),
+            cct = meta.request.getHeader(CaramelCompiledTemplates);
+
         js = js || [];
         css = css || [];
         code = code || [];
 
+        /**
+         * Set the Caramel-Compiled header to compile the required partial from the backend
+         * and send the HTML block to the frontend.
+         */
+        if (cct) {
+            areas = parse(cct);
+            if (areas != null) {
+                /**
+                 * At the moment we only support one key-value pair of partials to be rendered using this function
+                 * at once. Therefore retrieve the 0th element of the array.
+                 */
+                var areaKey = Object.keys(areas)[0];
+                path = caramel.theme().resolve(partialsDir + '/' + areas[areaKey] + '.hbs');
+                if (log.isDebugEnabled()) {
+                    log.debug('Rendering page : ' + path);
+                }
+                try {
+                    file = new File(path);
+                    file.open('r');
+                    template = Handlebars.compile(file.readAll());
+                } catch (e) {
+                    log.error(e.toString());
```

Comment:
- The purpose of logging the error here is for debugging purposes, not just for recording. So If we encounter any problem, we need get as much as information in logs to find out real cause of the issue.
- Since this method is called by FE js, we do need to log all information here. there is no other place to log the error. 
- logging exceptions using either getMessage() or toString(), don't always capture all information needed to pinpoint the problem. 
- If you are logging exception, you should log as much as you can. So that it provide enough information for a developer to discover what went wrong, where it went wrong, and why it went wrong.
Hope you get our point here.